publishing-sdk/src/main/java/com/ably/tracking/publisher/workerqueue/results/WorkResult.kt

```diff
@@ -0,0 +1,99 @@
+package com.ably.tracking.publisher.workerqueue.results
+
+import com.ably.tracking.ConnectionException
+import com.ably.tracking.TrackableState
+import com.ably.tracking.common.ConnectionStateChange
+import com.ably.tracking.common.PresenceMessage
+import com.ably.tracking.common.ResultCallbackFunction
+import com.ably.tracking.publisher.Trackable
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+internal typealias AsyncWork = (suspend () -> WorkResult)
+
+/**
+ * This sealed class represents a base result from concrete  [Worker] instances after they finish doing their work.
+ * It's also intended to be returned by [AsyncWork] coroutines.
+ * **/
+internal sealed class WorkResult
+
+/**
+ * A special [WorkResult] that contains an optional [syncWorkResult] and [asyncWork]
+ *
+ * @param syncWorkResult : Any work result that has resulted from synchronous work. This work result is expected to be
```

Comment:
Minor thing but `@param` syntax differs from the argument below. I would change to:

```suggestion
 * @param syncWorkResult any work result that has resulted from synchronous work. This work result is expected to be
```

Reply:
I've gone through the docs and aligned their style to match it project-wise https://github.com/ably/ably-asset-tracking-android/pull/607/commits/afb536853418b5bfdbdf9691ebe0b0737536adda